fennel/datasets/test_invalid_dataset.py

```diff
@@ -220,6 +221,77 @@ class RatingActivity:
     t: datetime
 
 
+def test_incorrect_assign_expr_type():
+    with pytest.raises(TypeError) as e:
+
+        @meta(owner="test@test.com")
+        @dataset
+        class RatingActivityTransformed:
+            userid: int
+            rating_sq: float
+            movie_suffixed: str
+            t: datetime
+
+            @pipeline
+            @inputs(RatingActivity)
+            def transform(cls, rating: Dataset):
+                return rating.assign(
+                    rating_sq=(F("rating") * F("rating")).astype(str),
+                    movie_suffixed=F("movie").str.concat("_suffix").astype(int),
+                ).drop("rating","movie")
+
+    assert (
+        str(e.value)
+        == """invalid assign - '[Pipeline:transform]->assign node' type errors: [TypeError("expression `(Field[`rating`] * Field[`rating`])` has type `<class 'float'>` does not match the type of column `rating_sq`, <class 'str'>"), TypeError('expression `Field[`movie`] + "_suffix"` has type `<class \\\'str\\\'>` does not match the type of column `movie_suffixed`, <class \\\'int\\\'>')]"""
+    )
+    
+    with pytest.raises(TypeError) as e2:
+
+        @meta(owner="test@test.com")
+        @dataset
+        class RatingActivityTransformed2:
+            userid: int
+            rating_sq: int
+            movie_suffixed: str
+            t: datetime
+
+            @pipeline
+            @inputs(RatingActivity)
+            def transform(cls, rating: Dataset):
+                return rating.assign(
+                    rating_sq=(F("rating") * F("rating")).astype(float),
+                    movie_suffixed=F("movie").str.concat("_suffix").astype(str),
+                ).drop("rating","movie")
+
+    assert (
+        str(e2.value)
+        == """[TypeError('Field `rating_sq` has type `float` in `pipeline transform output value` schema but type `int` in `RatingActivityTransformed2 value` schema.')]"""
+    )
+    
+    with pytest.raises(ValueError) as e2:
+
+        @meta(owner="test@test.com")
+        @dataset
+        class RatingActivityTransformed2:
+            userid: int
+            rating_sq: int
+            movie_suffixed: str
+            t: datetime
+
+            @pipeline
+            @inputs(RatingActivity)
+            def transform(cls, rating: Dataset):
+                return rating.assign(
+                    rating_sq=(F("rating") % F("rating")).astype(float),
+                    movie_suffixed=(F("movie") + "_suffix").astype(str),
+                ).drop("rating","movie")
+
+    assert (
+        str(e2.value)
+        == """invalid assign - '[Pipeline:transform]->assign node' error in expression for column `movie_suffixed`: Failed to compile expression: invalid expression: both sides of '+' must be numeric types but found String & String, left: col(movie), right: lit(String("_suffix"))"""
```

Comment:
I reckon this error is coming from Rust lib? If so, would strongly suggest to make sure that both Rust/Python print the same expression. For instance, this one uses `col` (with round brackets) and `String` but Python version uses `Field` with square brackets and `str`. Can we not use the print from the rust library itself in Python so there is a single copy of this?